frontend/fleet/helpers.ts

```diff
@@ -410,6 +412,75 @@ export const formatPackForClient = (pack: any): any => {
   return pack;
 };
 
+export const generateRole = (
+  teams: ITeam[],
+  globalRole: string | null
+): string => {
+  if (globalRole === null) {
+    const listOfRoles: any = teams.map((team) => team.role);
+
+    if (teams.length === 0) {
+      // no global role and no teams
+      return "Unassigned";
+    } else if (teams.length === 1) {
+      // no global role and only one team
+      return stringUtils.capitalize(teams[0].role ?? "");
+    } else if (
+      listOfRoles.every((role: string): boolean => role === "maintainer")
+    ) {
+      // only team maintainers
+      return stringUtils.capitalize(teams[0].role ?? "");
+    } else if (
+      listOfRoles.every((role: string): boolean => role === "observer")
+    ) {
+      // only team observers
+      return stringUtils.capitalize(teams[0].role ?? "");
+    }
```

Comment:
Can this be simplified?

```suggestion
    } else if (
      listOfRoles.every((role: string): boolean => role === "maintainer")
    ) {
      // only team maintainers
      return "Maintainer"
    } else if (
      listOfRoles.every((role: string): boolean => role === "observer")
    ) {
      // only team observers
      return "Observer"
    }
```

Reply:
Good catch, I made old code a helpers function but I didn't consider cleaning it up! 😅